state/validation_test.go

```diff
@@ -6,12 +6,22 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/zarbchain/zarb-go/block"
 	"github.com/zarbchain/zarb-go/crypto"
+	"github.com/zarbchain/zarb-go/crypto/bls"
+	"github.com/zarbchain/zarb-go/crypto/hash"
 	"github.com/zarbchain/zarb-go/sortition"
 	"github.com/zarbchain/zarb-go/tx"
 	"github.com/zarbchain/zarb-go/util"
 	"github.com/zarbchain/zarb-go/validator"
 )
 
+func aggregate(sigs []crypto.Signature) *bls.Signature {
+	blsSigs := make([]*bls.Signature, len(sigs))
+	for i, s := range sigs {
+		blsSigs[i] = s.(*bls.Signature)
+	}
+	return bls.Aggregate(blsSigs)
+}
+
```

Comment:
nice But I think `crypto.Singature` it self be able to handle this rather than consumer to be aggregating

Reply:
Unfortunately, Only BLS has this functionality. we can't use schnorr since it has nonce.